common_factories.py

```diff
@@ -30,6 +30,17 @@ def getQuickBuildFactory(mtrDbPool):
         autoCreateTables=True,
         env=MTR_ENV,
     ))
+    f_quick_build.addStep(steps.MTR(
+        logfiles={"mysqld*": "/buildbot/mysql_logs.html"},
+        command=["sh", "-c", util.Interpolate("cd mysql-test && exec perl mysql-test-run.pl --verbose-restart --force --retry=3 --max-save-core=1 --max-save-datadir=1 --max-test-fail=20 --mem --parallel=$(expr %(kw:jobs)s \* 2) --suite=wsrep,galera,galera_3nodes,galera_3nodes_sr %(kw:mtr_additional_args)s", mtr_additional_args=util.Property('mtr_additional_args', default=''), jobs=util.Property('jobs', default='$(getconf _NPROCESSORS_ONLN)'))],
+        timeout=600,
+        haltOnFailure="true",
+        parallel=mtrJobsMultiplier,
+        dbpool=mtrDbPool,
+        autoCreateTables=True,
+        env=MTR_ENV,
+        doStepIf=hasGalera
+    ))
```

Comment:
the only thing I am unsure about is how logs (mysqld files) are affected by multiple sequential runs of MTR. Do we need separate saving of the files between the two MTR runs?

Reply:
right, I'll look at how to combine to a single step

Reply:
I tried to combine to a single step. There was no mtr option that would continue on and append to the existing mysqld.X.err logs. So there's now a save step between the two MTR runs and a move of the main mtr tests to a different directory. I don't know about the CR interaction or parsing on these.

I added an additional check that the `$WSREP_PROVIDER` file exists. Pointing this to a non-existent file will fail badly:
```
$ WSREP_PROVIDER=/tmp/nothere mysql-test/mtr --suite=galera
Logging: /home/dan/repos/mariadb-server-10.4/mysql-test/mysql-test-run.pl  --suite=galera
VS config: 
vardir: /home/dan/repos/build-mariadb-server-10.4/mysql-test/var
Checking leftover processes...
Removing old var directory...
Creating var directory '/home/dan/repos/build-mariadb-server-10.4/mysql-test/var'...
Checking supported features...
MariaDB Version 10.4.29-MariaDB
 - SSL connections supported
Using suites: galera
Collecting tests...
 - binaries built with wsrep patch
mysql-test-run: *** ERROR: WSREP_PROVIDER env set to an invalid path
```

This should allow the extension of galera tests once the base stuff is working.